### PR TITLE
Fix autolinker

### LINKED ISSUE #234: Seeing ___ERB0___ for ERB within HTML
Let's say I have code like this in a diff or in a comment,

`<div class="image" style="<%= render(partial: '...') %>">`

What I end up seeing in Bitbucket is

`<div class="image" style="___ERB0___">`

In my code review, I actually asked, what is `___ERB0___` supposed to be? And then they responded with a comment saying they didn't know what was wrong, because it was supposed to say [pasted code]... that then rendered to the same thing with `___ERB0___` again, which is when we realized it was the extension getting confused.

### LINKED ISSUE #254: Links being injected and escaped
Suddenly I'm having an issue in PR review where email addresses, say in a repo url, are being expanded into html.

Without extension
```
-    def sourceRepo = "git@bitbucket.org:team/package.git"
```
With extension
```
-   def sourceRepo = "<a class="token email-link" href="mailto:git@bitbucket.org">git@bitbucket.org</a>:team/package.git"
```

I'd guess this is related to #245

### LINKED ISSUE #264: Syntax Highlighter strips out percent and some digits (%010d → %0d)
This picture features the same commit (in a pull request) twice. The above is _without_ syntax highlighting enabled, below is _with_ syntax highlighting enabled.

I use this extension mostly to assist in PR reviews, so this is kind of a dealbreaker bug.

Viewing a java file using Firefox 63 on Ubuntu, if it matters for reproduction.

### LINKED ISSUE #303: Syntax highlighter breaking PHP variables in a PHTML file
I work on Magento, which uses PHTML files for templates (essentially it's a PHP file that has a fancy extension).

I was just reviewing some code, and was surprised to see the following

RefinedBitbucket - Broken

In the image above, we can see `___PHP0___`, which I flagged in a comment to the author of the PR. Naturally I got a very confused response from the author.

Disabling Refined Bitbucket fixed up the code display to

RefinedBitbucket - Disabled

We can see the biggest difference between those 2 images (aside from the obvious odd variable), is the Syntax Highlighting. Sure enough, re-enabling Refined Bitbucket, and disabling just Syntax Highlighting will prevent this bug from occurring

### LINKED ISSUE #331: RB decodes encoded HTML entities
See this PR: https://bitbucket.org/gronostaj/html-escaping-test/pull-requests/1/expect-bugs-here/diff

The actual 2nd line is:

```
<?= '%7B%7Btest%7D%7D' =>
```

With RB enabled it looks like this:

```
<?= '{{test}}' =>
```

### LINKED ISSUE #335: Syntax Highlighter corrupt the display (`%X` is removed)
Issue found using the plugin to review C code.

The code `%XX` is removed from the display as you can see bellow:
image

You can see it on this PR:
https://bitbucket.org/clement-marrast/test/pull-requests/1/change-print-value/diff

Reproduced on Firefox 73.0.1 under Windows and plugin 3.20.

### PR DESCRIPTION
**Pull request by jwallet**
- don't decode url just link it
- don't parse partial url as url (actual fix)
- added support for data scheme (data:image, data:text, ?base64)

-   [ ] I updated the CHANGELOG.md
-   [x] I tested the changes in this pull request myself
-   [ ] I added Automated Tests
-   [ ] I added an Option to enable / disable this feature
-   [ ] I updated the README.md, with pictures if necessary

https://bitbucket.org/walletj/test/commits/64480cc901aef941c3095f580665e23ca2f467cb?w=1

---
the filename actually does not matter for autolinker, that's why i created a .js file and pasted every previous error rbb users got 

image

in the screenshot, links are in blue and other lines are reported strings in the issues below that parsed/converted by autolink and they render fine now
replace `//  #355` by `// #335` in the screenshot

Closes #234
Closes #254
Closes #264
Closes #303
Closes #331
Closes #335

### CODE REVIEW

File: src/vendor/prism-autolinker.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+;(function() {
```
Comment (jwallet) [inline src/vendor/prism-autolinker.js:1]: i minified my version of autolink in the file below

**Comment by jwallet** [comment]
hey guys @reyronald  @andremw let me know if you guys need more mainteners for this project, I think I might work on this extension for quite some times since I use it at work and Bitbucket Cloud is kind of a mess. I will like to set up continuous integration tool, automated tests for prs using appveyor so we don't merge failed unit tests by mistake, it's quick and reliable, or setting up a board to leave notes or let you know what I'm working on, for instance currently i'm on custom/default reviewers

**Comment by reyronald** [comment]
CI is already set-up with Travis, you can check it out here: https://travis-ci.org/github/refined-bitbucket/refined-bitbucket

It was running automatically with PRs and commits before but apparently it was de-activated somehow, I'll re-activate it again, thanks for pointing that out

Setting up a board seems overkill to be honest, I don't think we need that.

**Comment by andremw** [comment]
Hey, @jwallet, it's been a long time since I stopped using BitBucket (and the extension) but @reyronald has done a brilliant work so far, so he's the one who will be able to bring you aboard as an official maintainer.
I try to read everything you do here to learn a little bit and I must say that it's been a pleasure for me to see such an active contributor. You're doing great! I hope you stay longer with us!

File: src/vendor/prism-autolinker.js (hunk 1 above)
Comment (reyronald) [reply src/vendor/prism-autolinker.js:1]: Did you modify this file yourself or is this an updated version of the autolinker plugin from Prism?

**Review by reyronald — APPROVED** [review]

**Pull request merged**